BUG: fix to_latex bold_rows option

kdebrab opened the pull request:
- [x] closes #16707
 - [x] tests added / passed
 - [ ] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

kdebrab committed:
BUG: fix to_latex bold_rows option

jreback commented on pandas/io/formats/format.py:
```diff
@@ -943,6 +943,11 @@ def get_col_type(dtype):
                          if x else '{}') for x in row]
             else:
                 crow = [x if x else '{}' for x in row]
+            if self.fmt.kwds.get('bold_rows', True) and self.fmt.index:
```
in the constructor do a ``self.bold_rows = self.fmt.kwd.get('bold_rows', True)``; see ``HTMLFormatter``. 

Then just use this here

jreback reviewed (changes requested):
looks reasonable. pls add a bug fix whatsnew entry (in IO) for 0.20.3

kdebrab commented:
I had to adapt  the expected result of all existing tests where. to_latex() was used, as the default behaviour changed. I'm not so sure though whether it's a good idea to set bold_rows to True by default. I guess the reason for this documented  default is the analogy with. to_html(), but in that output also the column headers are bold. In addition, bold row and/or column headers do not seem very common in latex community examples.

chris-b1 commented:
Could you show an example what the before/after output looks like rendered?  I'm inclined to agree that changing the default isn't a good idea.

kdebrab commented:
With `bold_rows=False` (i.e. 'before' output):
![image](https://user-images.githubusercontent.com/1150402/27233171-daa1846a-52b8-11e7-9fdb-d9abb439e3b5.png)
With `bold_rows=True` (would be 'after' default output):
![image](https://user-images.githubusercontent.com/1150402/27233205-02a8be88-52b9-11e7-8c93-90b033b67907.png)

It's a matter of taste, but I don't like the latter one that much...

I'm in favor of keeping the default output the same as before, and correcting the signature (and docstring) of .to_latex() accordingly.

TomAugspurger commented:
> I'm in favor of keeping the default output the same as before, and correcting the signature (and docstring) of .to_latex() accordingly.

+1

kdebrab commented:
@jreback: As proposed, I made sure to have the same default behaviour as before (i.e. `bold_rows=False`) and corrected the `.to_latex()` signature accordingly.

I also added and updated a whatsnew entry as you asked, so I guess this is ready to be merged.

jreback commented:
lgtm. can you rebase. ping on green. @TomAugspurger

TomAugspurger commented on doc/source/whatsnew/v0.20.3.txt:
```diff
@@ -55,7 +54,8 @@ Indexing
 
--- Bug in ``pd.read_csv()`` in which files weren't opened as binary files by the C engine on Windows, causing EOF characters mid-field, which would fail (:issue:`16039`, :issue:`16559`, :issue`16675`)
+- Bug in ``pd.read_csv()`` in which files weren't opened as binary files by the C engine on Windows, causing EOF characters mid-field, which would fail (:issue:`16039`, :issue:`16559`, :issue`16675`)
+- Bug in ``.to_latex()`` where ``bold_rows`` was wrongly specified to be True by default, whereas in reality row labels remained non-bold whatever parameter provided. (:issue:`16707`)
```
Maybe just say here that the default behavior is the same (not bold), but can now be overridden correctly.

TomAugspurger reviewed (approved):
Looks good, small suggestion to clarify the whatsnew that you can change as you rebase.

jreback commented:
thanks!
